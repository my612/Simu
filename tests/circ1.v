module circ1(in0, in1, in2, in3 , sel1, sel0, out);

input in0;
input in1;
input in2;
input in3;
input sel1;
input sel0;

output out;

wire w1;
wire w2;
wire w3;
wire w4;
wire w5;
wire w6;
wire w7;
wire w8;
wire w9;
wire w10;

wire sel_not0;
wire sel_not1;

not #(2) g1 (sel_not0,sel0);
not #(2) g2 (sel_not1,sel1);


and #(5) g3 (w1,sel_not1,sel_not0);
and #(5) g4 (w2,w1,in0); 


and #(5) g5 (w3,sel_not1,sel0);
and #(5) g6 (w4,w3,in1); 


and #(5) g7 (w5,sel1,sel_not0);
and #(5) g8 (w6,w5,in2); 

and #(5) g9 (w7,sel1,sel0);
and #(5) g10 (w8,in3,w7);

or #(9) g11 (w9,w2,w4);
or #(9) g12 (w10,w6,w8);
or #(9) g13 (out,w9,w10);



endmodule;