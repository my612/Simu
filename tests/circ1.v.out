#! /usr/bin/vvp
:ivl_version "12.0 (stable)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision + 0;
:vpi_module "/usr/lib/x86_64-linux-gnu/ivl/system.vpi";
:vpi_module "/usr/lib/x86_64-linux-gnu/ivl/vhdl_sys.vpi";
:vpi_module "/usr/lib/x86_64-linux-gnu/ivl/vhdl_textio.vpi";
:vpi_module "/usr/lib/x86_64-linux-gnu/ivl/v2005_math.vpi";
:vpi_module "/usr/lib/x86_64-linux-gnu/ivl/va_math.vpi";
S_0x613aa69614f0 .scope module, "circ1" "circ1" 2 1;
 .timescale 0 0;
    .port_info 0 /INPUT 1 "in0";
    .port_info 1 /INPUT 1 "in1";
    .port_info 2 /INPUT 1 "in2";
    .port_info 3 /INPUT 1 "in3";
    .port_info 4 /INPUT 1 "sel1";
    .port_info 5 /INPUT 1 "sel0";
    .port_info 6 /OUTPUT 1 "out";
o0x7fab59da0108 .functor BUFZ 1, C4<z>; HiZ drive
L_0x613aa69b0510/d .functor NOT 1, o0x7fab59da0108, C4<0>, C4<0>, C4<0>;
L_0x613aa69b0510 .delay 1 (2,2,2) L_0x613aa69b0510/d;
o0x7fab59da0138 .functor BUFZ 1, C4<z>; HiZ drive
L_0x613aa69b0730/d .functor NOT 1, o0x7fab59da0138, C4<0>, C4<0>, C4<0>;
L_0x613aa69b0730 .delay 1 (2,2,2) L_0x613aa69b0730/d;
L_0x613aa69b08a0/d .functor AND 1, L_0x613aa69b0730, L_0x613aa69b0510, C4<1>, C4<1>;
L_0x613aa69b08a0 .delay 1 (5,5,5) L_0x613aa69b08a0/d;
o0x7fab59da0018 .functor BUFZ 1, C4<z>; HiZ drive
L_0x613aa69b0a80/d .functor AND 1, L_0x613aa69b08a0, o0x7fab59da0018, C4<1>, C4<1>;
L_0x613aa69b0a80 .delay 1 (5,5,5) L_0x613aa69b0a80/d;
L_0x613aa69b0c10/d .functor AND 1, L_0x613aa69b0730, o0x7fab59da0108, C4<1>, C4<1>;
L_0x613aa69b0c10 .delay 1 (5,5,5) L_0x613aa69b0c10/d;
o0x7fab59da0048 .functor BUFZ 1, C4<z>; HiZ drive
L_0x613aa69b0d70/d .functor AND 1, L_0x613aa69b0c10, o0x7fab59da0048, C4<1>, C4<1>;
L_0x613aa69b0d70 .delay 1 (5,5,5) L_0x613aa69b0d70/d;
L_0x613aa69b0f10/d .functor AND 1, o0x7fab59da0138, L_0x613aa69b0510, C4<1>, C4<1>;
L_0x613aa69b0f10 .delay 1 (5,5,5) L_0x613aa69b0f10/d;
o0x7fab59da0078 .functor BUFZ 1, C4<z>; HiZ drive
L_0x613aa69b1070/d .functor AND 1, L_0x613aa69b0f10, o0x7fab59da0078, C4<1>, C4<1>;
L_0x613aa69b1070 .delay 1 (5,5,5) L_0x613aa69b1070/d;
L_0x613aa69b1220/d .functor AND 1, o0x7fab59da0138, o0x7fab59da0108, C4<1>, C4<1>;
L_0x613aa69b1220 .delay 1 (5,5,5) L_0x613aa69b1220/d;
o0x7fab59da00a8 .functor BUFZ 1, C4<z>; HiZ drive
L_0x613aa69b1360/d .functor AND 1, o0x7fab59da00a8, L_0x613aa69b1220, C4<1>, C4<1>;
L_0x613aa69b1360 .delay 1 (5,5,5) L_0x613aa69b1360/d;
L_0x613aa69b1580/d .functor OR 1, L_0x613aa69b0a80, L_0x613aa69b0d70, C4<0>, C4<0>;
L_0x613aa69b1580 .delay 1 (9,9,9) L_0x613aa69b1580/d;
L_0x613aa69b16e0/d .functor OR 1, L_0x613aa69b1070, L_0x613aa69b1360, C4<0>, C4<0>;
L_0x613aa69b16e0 .delay 1 (9,9,9) L_0x613aa69b16e0/d;
L_0x613aa69b1900/d .functor OR 1, L_0x613aa69b1580, L_0x613aa69b16e0, C4<0>, C4<0>;
L_0x613aa69b1900 .delay 1 (9,9,9) L_0x613aa69b1900/d;
v0x613aa6961730_0 .net "in0", 0 0, o0x7fab59da0018;  0 drivers
v0x613aa69af680_0 .net "in1", 0 0, o0x7fab59da0048;  0 drivers
v0x613aa69af740_0 .net "in2", 0 0, o0x7fab59da0078;  0 drivers
v0x613aa69af7e0_0 .net "in3", 0 0, o0x7fab59da00a8;  0 drivers
v0x613aa69af8a0_0 .net "out", 0 0, L_0x613aa69b1900;  1 drivers
v0x613aa69af9b0_0 .net "sel0", 0 0, o0x7fab59da0108;  0 drivers
v0x613aa69afa70_0 .net "sel1", 0 0, o0x7fab59da0138;  0 drivers
v0x613aa69afb30_0 .net "sel_not0", 0 0, L_0x613aa69b0510;  1 drivers
v0x613aa69afbf0_0 .net "sel_not1", 0 0, L_0x613aa69b0730;  1 drivers
v0x613aa69afcb0_0 .net "w1", 0 0, L_0x613aa69b08a0;  1 drivers
v0x613aa69afd70_0 .net "w10", 0 0, L_0x613aa69b16e0;  1 drivers
v0x613aa69afe30_0 .net "w2", 0 0, L_0x613aa69b0a80;  1 drivers
v0x613aa69afef0_0 .net "w3", 0 0, L_0x613aa69b0c10;  1 drivers
v0x613aa69affb0_0 .net "w4", 0 0, L_0x613aa69b0d70;  1 drivers
v0x613aa69b0070_0 .net "w5", 0 0, L_0x613aa69b0f10;  1 drivers
v0x613aa69b0130_0 .net "w6", 0 0, L_0x613aa69b1070;  1 drivers
v0x613aa69b01f0_0 .net "w7", 0 0, L_0x613aa69b1220;  1 drivers
v0x613aa69b02b0_0 .net "w8", 0 0, L_0x613aa69b1360;  1 drivers
v0x613aa69b0370_0 .net "w9", 0 0, L_0x613aa69b1580;  1 drivers
# The file index is used to find the file name in the following table.
:file_names 3;
    "N/A";
    "<interactive>";
    "circ1.v";
